File: src/utils/add-commitlint.ts
// Third-Party libraries
import inquirer from 'inquirer'

// NodeJS
import process from 'node:process'

// Utils
import { writeMessage } from './console'
import { getErrorMessage } from './errors'

export async function addCommitlint(): Promise<boolean> {
  try {
    const { addCommitlint } = await inquirer.prompt({
      type: 'confirm',
      default: true,
      name: 'addCommitlint',
      message: 'Do you wanna add commitlint?:'
    })

    return addCommitlint
  } catch {
    writeMessage({
      type: 'error',
      message: getErrorMessage('CommitlintSelection')
    })
    process.exit(1)
  }
}

File: internal/user_input/add_commitlint.go
package user_input

import (
	"os"

	"github.com/AlecAivazis/survey/v2"
	"github.com/RaulCatalinas/HuskyBC/internal/enums"
	errorMessages "github.com/RaulCatalinas/HuskyBC/internal/error_messages"
	"github.com/RaulCatalinas/HuskyBC/internal/utils"
)

func AddCommitlint() bool {
	var questions = []*survey.Question{
		{
			Name: "addCommitlint",
			Prompt: &survey.Confirm{
				Message: "Do you wanna add commitlint?",
				Default: true,
			},
		},
	}

	answers := struct {
		AddCommitlint bool `survey:"addCommitlint"`
	}{}

	err := survey.Ask(questions, &answers)

	if err != nil {
		utils.WriteMessage(utils.WriteMessageProps{
			Type:    enums.MessageTypeError,
			Message: errorMessages.USER_INPUT_ERROR_MESSAGES[enums.AddCommitLintError],
		})

		os.Exit(1)
	}

	return answers.AddCommitlint
}
